aws/resource_aws_appsync_schema.go

```diff
@@ -0,0 +1,87 @@
+package aws
+
+import (
+	"errors"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/appsync"
+	"github.com/hashicorp/terraform/helper/schema"
+	"time"
+)
+
+func resourceAwsAppsyncSchema() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAppsyncSchemaPut,
+		Read:   resourceAwsAppsyncSchemaNil,
+		Update: resourceAwsAppsyncSchemaPut,
+		Delete: resourceAwsAppsyncSchemaNil,
+
+		Schema: map[string]*schema.Schema{
+			"api_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"definition": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAppsyncSchemaPut(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appsyncconn
+
+	apiId := d.Get("api_id").(string)
+	d.SetId(apiId)
+
+	if d.HasChange("definition") {
+		input := &appsync.StartSchemaCreationInput{
+			ApiId:      aws.String(apiId),
+			Definition: ([]byte)(d.Get("definition").(string)),
+		}
+		if _, err := conn.StartSchemaCreation(input); err != nil {
+			return err
+		}
+
+		if err := waitForSchemaToBeActive(apiId, meta); err != nil {
```

Comment:
@beauknowssoftware I've been working on implementing AppSync resolvers and came across your PR via #2705 . I'm wondering if this `wait` could be implemented using `StateChangeConf` provided in terraform/helper/resource. Here's an example in CloudFormation stacks: https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_cloudformation_stack.go#L178. I believe it can since there are discrete statuses returned from `GetSchemaCreationStatus`

Reply:
I will agree with @perryao here -- we tend to prefer API waiting via existing SDK `Waiter` functions (pretty rare) and either `resource.Retry()` or `resource.StateChangeConf`: https://www.terraform.io/docs/extend/resources.html#statechangeconf

There are quite a few examples across the project if you search for `StateChangeConf`, but I can certainly help if you need specific guidance. 😄